src/doc/unstable-book/src/language-features/loop-break-value.md

```diff
@@ -4,7 +4,76 @@ The tracking issue for this feature is: [#37339]
 
 [#37339]: https://github.com/rust-lang/rust/issues/37339
 
+Documentation to be appended to section G of the book.
+
 ------------------------
 
+### Loops as expressions
+
+Like most things in Rust, loops are expressions, and have a value; normally `()` unless the loop
+never exits.
```

Comment:
Nice, I think this is more to the point now. "unless the loop never exits" only applies to `loop` and not any loop that never exists (see how even we are getting confused by the terms), `for` and `while` _always_ evaluate to `()`. But diverging things are very niche, maybe we should just not mention this here?